airflow-core/pyproject.toml

```diff
@@ -135,8 +135,8 @@ dependencies = [
     "rich-argparse>=1.0.0",
     "rich>=13.6.0",
     "setproctitle>=1.3.3",
-    # The issue tracking deprecations for sqlalchemy 2 is https://github.com/apache/airflow/issues/28723
-    "sqlalchemy[asyncio]>=1.4.49",
+    # SQLAlchemy >=2.0.36 fixes Python 3.13 TypingOnly import AssertionError caused by new typing attributes (__static_attributes__, __firstlineno__)
+    "sqlalchemy[asyncio]>=2.0.36",
```

Comment:
Does it mean with this PR we force upgrading to sqlalchemy 2.x and do _not_ support 1.x anymore? Is this the desired state or do we leave integrations behind which still rely on 1.x?
(I was not involved in the discussions, so just asking if we want to force it then with Airflow 3.2)

Reply:
Yes. I think sqlalchemy 1 should be dropped. I think we should have a short DISCUSS thread on it with pro/cons from people who have opinions and have a vote/consensus on it

Reply:
Had also one thought in my mind - I think we are not affected but not sure how much custom code could be affected relying on SQLalchemy? Airflow <=3.1 is pinned to require SQLA<2 and if we merge it like this then Airflow >=3.2 will require SQLA>=2.

In Pydantic we also had such breaking change, maybe there are more people affected but there also we supported both v1+v2 in parallel for a while. (yes and with the overhead having a CI test always running Pydantic v1 as well...)